### PR TITLE
Add further Scout CI guides

### PR DESCRIPTION
**Pull request by ChrisChinchilla**

### CODE REVIEW

File: scout/ci.md [hunk 1]
```diff
@@ -140,10 +141,230 @@ Add the following to the YAML file:
...
+        displayName: Build
+        pool:
+          vmImage: ubuntu-latest
+        steps:
+          - task: Docker@2
+            displayName: Build an image
+            inputs:
+              command: build
+              dockerfile: "$(Build.SourcesDirectory)/Dockerfile"
+              repository: $(image)
```
Comment (ChrisChinchilla) [inline scout/ci.md:264]: What repository is this?

Comment (eunomie) [reply scout/ci.md:264]: my guess is it's the `variables.image` defined above (so here set to `vonwig/nodejs-service`)

File: scout/ci.md [hunk 2]
```diff
@@ -140,10 +141,230 @@ Add the following to the YAML file:
...
+```
+
+This defines the container image the workflow uses and an environment variable for the image.
+
+Add the following to the YAML file to define the steps for the workflow:
+
+```yaml
+steps:
+  # Checkout the repository files
+  - checkout
```
[the remark this answers is not in the document]
Comment (eunomie) [reply scout/ci.md:309]: This is a special step from `circleci`: https://circleci.com/docs/configuration-reference#checkout

File: scout/ci.md (hunk 1 above)
Comment (ChrisChinchilla) [reply scout/ci.md:264]: @eunomie That I get, but was more meaning what's the difference between the image defined in the Dockerfile and the repository mentioned?

Comment (cdupuis) [reply scout/ci.md:264]: That's the same. `repository` is the parameter name for that step which takes an image name. The value comes from the variable as @eunomie said. I only extracted the repository into the `image` var to be able to reuse this later in the `docker scout cves $(image)` command.

**Review by cdupuis — APPROVED** [review]

File: scout/ci.md [hunk 3]
```diff
@@ -140,10 +141,255 @@ Add the following to the YAML file:
...
+    - |
+      apk add --update curl
+      curl -sSfL https://raw.githubusercontent.com/docker/scout-cli/main/install.sh | sh -s -- 
+      apk del curl 
+      rm -rf /var/cache/apk/*
+    # Login to Docker Hub required for Docker Scout CLI
+    - docker login -u "$DOCKER_HUB_USER" -p "$DOCKER_HUB_PAT"
+```
+
+This sets up the workflow to build Docker images with GitLabs's
```
Comment (usha-mandya) [inline scout/ci.md:185]: nit

```suggestion
This sets up the workflow to build Docker images with GitLab's
```

File: scout/ci.md [hunk 4]
```diff
@@ -140,10 +141,255 @@ Add the following to the YAML file:
...
+"Docker-in-Docker" mode to run Docker inside a container.
+
+It then downloads curl and the Docker CLI and logs into the Gitlab CI registry
```
Comment (usha-mandya) [inline scout/ci.md:188]:
```suggestion
It then downloads curl and the Docker CLI and logs into the GitLab CI registry
```

**Review by usha-mandya — APPROVED** [review]

**Pull request merged**